kv/memberlist/kv_init_service.go

```diff
@@ -171,21 +171,13 @@ func (kvs *KVInitService) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if strings.Contains(accept, "application/json") {
 		w.Header().Set("Content-Type", "application/json")
 
-		data, err := json.Marshal(v)
-		if err != nil {
+		if err := json.NewEncoder(w).Encode(v); err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
 		}
-
-		// We ignore errors here, because we cannot do anything about them.
-		// Write will trigger sending Status code, so we cannot send a different status code afterwards.
-		// Also this isn't internal error, but error communicating with client.
-		_, _ = w.Write(data)
 		return
 	}
 
-	err := defaultPageTemplate.Execute(w, v)
-	if err != nil {
+	if err := defaultPageTemplate.Execute(w, v); err != nil {
```

Comment:
We should set `Content-Type` in this branch too.

Reply:
Added it here: https://github.com/grafana/dskit/pull/149/commits/d2926ce863a81aad855a73401b9069c03f8d219f

Reply:
Thank you.